pulser/simulation/simulation.py

```diff
@@ -210,36 +210,20 @@ def _build_basis_and_op_matrices(self) -> None:
                 self.basis[proj[0]] * self.basis[proj[1]].dag()
             )
 
-    def _build_operator(self, op_id: str, *qubit_ids: Union[str, int],
-                        global_op: bool = False) -> qutip.Qobj:
+    def _build_operator(self, operations: dict,
+                        ) -> qutip.Qobj:
         """Create qutip.Qobj with nontrivial action at *qubit_ids."""
-        if global_op:
-            return sum(self._build_operator(op_id, q_id)
-                       for q_id in self._qdict)
-        if len(set(qubit_ids)) < len(qubit_ids):
-            raise ValueError("Duplicate atom ids in argument list.")
-        # List of identity operators, except for op_id where requested:
-        op_list = [self.op_matrix[op_id]
-                   if j in map(self._qid_index.get, qubit_ids)
-                   else self.op_matrix['I'] for j in range(self._size)]
-        return qutip.tensor(op_list)
-
-    def _build_general_operator(
-        self, op_id: list[str], qubit_ids: list[Union[str, int]]
-            ) -> qutip.Qobj:
-        """Create qutip.Qobj with nontrivial actions at *qubit_ids.
-        op_id and qubits_id are a list of same length. op_id[j]
-        acts on qubits_id[j]"""
-        if len(set(qubit_ids)) < len(qubit_ids):
-            raise ValueError("Duplicate atom ids in argument list.")
-        if len(op_id) != len(qubit_ids):
-            raise ValueError(
-                "Different number of operators and qubits")
-        # List of identity operators, except for op_id where requested:
         op_list = [self.op_matrix['I'] for j in range(self._size)]
-        for j, qubit in enumerate(qubit_ids):
-            k = self._qid_index[qubit]
-            op_list[k] = self.op_matrix[op_id[j]]
+        for operator, qubits in operations.items():
```

Comment:
This is looking good! Can you extend this to the case when `operator` can be a `Qobj` and `qubits` are given as labels with prefix (e.g. `'atom4'`)?